predicates/query/query.go

```diff
@@ -0,0 +1,110 @@
+/*
+Package source implements a custom predicate to match routes
+based on the Query Params in URL
+
+It supports checking existence of query params and also checking whether
+query params value match to a given regular exp
+
+Examples:
+
+    // Checking existence of a query param
+    // matches http://example.org?bb=a&query=withvalue
+    example1: QueryParam("query") -> "http://example.org";
+
+    //Even a query param without a value
+    // matches http://example.org?bb=a&query=
+    example1: QueryParam("query") -> "http://example.org";
+
+    //matches with regexp
+    // matches http://example.org?bb=a&query=example
+    example1: QueryParam("query", "^example$") -> "http://example.org";
+
+    //matches with regexp and multiple values of query param
+    // matches http://example.org?bb=a&query=testing&query=example
+    example1: QueryParam("query", "^example$") -> "http://example.org";
+
+*/
+
+package query
+
+import (
+	"github.com/zalando/skipper/predicates"
+	"github.com/zalando/skipper/routing"
+	"net/http"
+	"regexp"
+)
+
+type MatchType int
+
+const (
+	Exists  MatchType = 1
+	Matches MatchType = 2
```

Comment:
while correct, this can be written somewhat nicer:

```
type MatchType int

const (
    Exists  MatchType = iota
    Matches
)
```

iota is a go idiom for these kind of constants.